.github/workflows/npm.yml

```diff
@@ -0,0 +1,90 @@
+# alternative KISS pipeline to bloated ci.yml, modeled after vscode-ansible one.
+name: npm
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  FORCE_COLOR: "1" # make mocha output colorful
+  # https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/
+  WSLENV: FORCE_COLOR:GITHUB_PATH/p
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  test:
+    name: ${{ matrix.name || format('{0} ({1})', matrix.npm-target, matrix.os) }}
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.shell || 'bash'}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-11
+        npm-target:
+          - test
+        upload-artifact:
+          - false
+        name:
+          - false
+        include:
+          - name: test (wsl)
+            npm-target: test
+            # https://github.com/actions/virtual-environments/issues/5151
+            os: windows-2019
```

Comment:
That is needed as windows-2022 wsl seems unstable. Read https://github.com/actions/virtual-environments/issues/5151